compiler/rustc_middle/src/traits/mod.rs

```diff
@@ -259,7 +259,7 @@ pub enum ObligationCauseCode<'tcx> {
 
     BuiltinDerivedObligation(DerivedObligationCause<'tcx>),
 
-    ImplDerivedObligation(DerivedObligationCause<'tcx>),
+    ImplDerivedObligation(Box<ImplDerivedObligationCause<'tcx>>),
```

Comment:
try using Lrc instead of Box like it's being used on the field of DerivedObligationCause

Reply:
Done